### ISSUE TITLE
Failed to convert device from layer3 to hybrid

### ISSUE THREAD

**Issue by cromero-packet-ph** [terraform-providers/terraform-provider-packet#224]
### Briefing

We're seeing this error `Error: Failed to convert device <instance_id> from layer3 to hybrid. New type was layer3` when adding the `network_type = "hybrid"` parameter to the terraform script we're using to provision an n2.xlarge.x86 server type, as derived from the documentation here: https://www.terraform.io/docs/providers/packet/r/device.html

### Terraform Version
_Tested on both versions:_
Terraform v0.12.12
Terraform v0.12.21

### Affected Resource(s)
packet_device

### Terraform Configuration Files
```hcl
provider "packet" {
  auth_token = "api_token"
}

locals {
  project_id = "my_project_id"
}

resource "packet_device" "n2xl-esxi" {
  hostname         = "n2xl-00001"
  plan             = "n2.xlarge.x86"
  facilities	   = ["ewr1"]
  operating_system = "vmware_esxi_6_5"
  billing_cycle    = "hourly"
  public_ipv4_subnet_size = "29"
  tags             = ["case-esxi-n2xl"]
  project_id       = local.project_id
  network_type	   = "hybrid"
}
```
### Expected Behavior
Instance will be created and provisioned successfully, with "Networking Type" set from **Layer 3** to **Hybrid**.

### Actual Behavior
Instance created and provisioned, with "Networking Type" set from **Layer 3** to **Hybrid** as verified from Packet's Customer Portal, but the error below shows after terraform operation completed: 

`Error: Failed to convert device <instance_id> from layer3 to hybrid. New type was layer3`

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. modified the terraform script file, added the `network_type = "hybrid"` configuration
2. saved the .tf file, run `terraform validate` to confirm parameters were correct
3. executed `terraform apply`

**Comment by t0mk** [comment]
hey @cromero-packet-ph, thanks for the PR. What is your provider version? Latest is v2.7.5. You can check it with `$ terraform version`

**Comment by t0mk** [comment]
Can you please run it again, this time, run the apply like 
```
TF_LOG=DEBUG TF_LOG_PATH=tflog  terraform apply -auto-approve
```

.. and upload the tflog file? Before uploading, please check that your auth token was blanked from the HTTP logs.

**Comment by cromero-packet-ph** [comment]
> hey @cromero-packet-ph, thanks for the PR. What is your provider version? Latest is v2.7.5. You can check it with `$ terraform version`

Here's the output of `$ terraform version`:

`Terraform v0.12.21`
`+ provider.packet v2.5.0`

Guess need to run a provider update then trying it again.

**Comment by cromero-packet-ph** [comment]
@t0mk here's the gist for the debug log for your inspection:
https://gist.github.com/cromero-packet-ph/e912dc91bdc57c71ae8d0be6cc0e47b0

**Comment by cromero-packet-ph** [comment]
I ran a provider update with my system, showing the latest `$terraform version`:

`Terraform v0.12.21`
`+ provider.packet v2.7.5`

Due to big data size issue, I've resoluted to create a GH repo where I can stash the latest debug log, with tokens and other obvious sensitive data removed:
https://github.com/cromero-packet-ph/stash

**Comment by t0mk** [comment]
@cromero-packet-ph Thanks. There's sth wrong with the eth and bond ports order.

I think something has changed in the API, I don't remember having two bond ports ever. The network type of the device has been implemented as the network_type of the `bond0` interface:
https://github.com/packethost/packngo/blob/1156d996f0d5fde76130da82f29abfa70b846c6e/devices.go#L116

In your HTTP log I can see that the network_type is listed on both bond0 and bond1, and bond1 is indeed in the hybrid network state, but the Golang SDK reports the network state from the bond0.

For some reason (which I don't remember anymore), eth1 is used in packgno for disbonding when changing network_type to hybrid:
https://github.com/packethost/packngo/blob/1156d996f0d5fde76130da82f29abfa70b846c6e/ports.go#L230

.. and that probably causes that it's the bond1 in hybrid mode and not bond0. @cromero-packet-ph do you think you could bring attention of someone who's aware of the correct procedure of going from layer3 to hybrid on n2.xlarge.x86? I would need to know
- which eth port should be disbonded (eth0 or eth1)
- which bond port (bond0 or bond1) shows the network_type of the device. Both of those ports have `network_type` attributes.

**Comment by t0mk** [comment]
I created a n2.xlarge device today, and then put it to "hybrid" mode. I checked how the ports look over the API and it was:
- bond0 had "network_type": "layer3"
- bond1 had "network_type": "layer2-individual"
- eth0 [in bond0] was bonded
- eth1 [in bond1] was not bonded
- eth2 [in bond0] was bonded
- eth3 [in bond1] was not bonded

So it's a bit different state than the other plans with only one bond, for example m1.xlarge. If I put an m1.xlarge device to "hybrid" mode, the situation is
- bond0 has network_type "hybrid" (it's the only bond)
- eth0 [in bond0] is bonded
- eth1 [in bond0] is not bonded

Is the hybrid mode different in the n2.xlarge plan (which has 2 bonds)? I can make the Golang SDK work with that. Are there any other plans that need special treatment?

**Comment by t0mk** [comment]
I wrote down network_type transitions in terms of API calls for n2.xlarge:


L3 -> hybrid:
 - /disbond API call to eth1
 - /disbond API call to eth3

hybrid -> L3:
 - /bond API call to eth1
 - /bond API call to eth3

 
L3 -> L2-individual
 - /convert/layer-2 API to bond0 
 - /disbond API call to bond1
 - /disbond API call to bond0

L2-individual -> L3
 - /bond API call to bond0
 - /bond API call to bond1
 - (only sometimes) /convert/layer3 API call to bond0. If issuerds Results in error "bond0 is already layer3"

(It's impossible to put n2.xlarge device to "layer2-bonded" state, it always goes/stays in "layer3")
L3 -> L2-bonded
 - /convert/layer-2 API call to bond0. Results to success but both bonds stay in "layer3".

hybrid -> L2-bonded
 - /convert/layer-2 API call to bond0 
 - /bond API call to eth1
 - /bond API call to eth3
 - All API calls are success but resulting netowrk type is Layer3

hybrid -> L2-individual
 - /convert/layer-2 API call to bond0 
 - /disbond API call to bond1
 - /disbond API call to bond0

L2-individual -> hybrid
 - /bond API call to bond0
 - /bond API call to bond1
 - /disbond API call to eth1
 - /disbond API call to eth3

**Comment by t0mk** [comment]
@cromero-packet-ph this should be fixed in provider version 2.8.0. Please update your provider and test. Let me know if it doesn't work. thanks!
